tests/integration/notifications.test.js

```diff
@@ -0,0 +1,466 @@
+import Web3 from 'web3'
+
+import Kleros from '../../src/kleros'
+import * as ethConstants from '../../src/constants/eth'
+import * as notificationConstants from '../../src/constants/notification'
+
+import { setUpContracts, waitNotifications, resetUserProfile } from './helpers'
+
+describe('Notifications and Event Listeners', () => {
+  let partyA
+  let partyB
+  let juror1
+  let juror2
+  let other
+  let web3
+  let KlerosInstance
+  let storeProvider
+  let notificationCallback
+  let notifications = []
+  let klerosPOCData
+  let arbitrableContractData
+  let klerosPOCAddress
+  let arbitrableContractAddress
+  let rngAddress
+  let pnkAddress
+
+  beforeAll(async () => {
```

Comment:
You don't really need to use `beforeAll`. You can just make the function passed to the parent `describe` `async` and do all of this inline.

Reply:
I think its still useful to have it here so if we want to add more tests to this file we don't need to pull it back out